plugins/ff_mpeg.py

```diff
@@ -1,156 +1,350 @@
+            file_name = replied.audio.file_name
+        elif hasattr(replied.video, 'file_name'):
+            file_name = replied.video.file_name
+        elif hasattr(replied.document, 'file_name'):
+            file_name = replied.document.file_name
         else:
-            await message.edit("Reply to a Telegram media file")
+            file_name = Path(dl_loc).name
+    else:
+        is_url = re.search(r"(?:https?|ftp)://[^|\s]+\.[^|\s]+", input_str)
+        if is_url:
+            try:
+                dl_loc, _ = await url_download(message, input_str)
+                file_name = Path(dl_loc).name
+            except Exception as e_e:
+                await message.err(e_e)
+                return
+    if dl_loc:
+        file_path = dl_loc
     else:
-        await message.edit(
-            "a media file already exists in path. "
-            f"Please remove the media and try again!\n`.term rm {FF_MPEG_DOWN_LOAD_MEDIA_PATH}`")
-
-
-@userge.on_cmd("ffmpegtrim", about={
-    'header': "Trim a given media",
-    'usage': "{tr}ffmpegtrim [start time] [end time]"})
-async def ffmpegtrim(message: Message):
-    if not os.path.exists(FF_MPEG_DOWN_LOAD_MEDIA_PATH):
-        await message.edit(
-            "a media file needs to be downloaded, and saved to the "
-            f"following path: `{FF_MPEG_DOWN_LOAD_MEDIA_PATH}`")
-        return
-    current_message_text = message.text
-    cmt = current_message_text.split(" ")
+        file_path = input_str.strip()
+        file_name = Path(file_path).name
+    if not Path(file_path).exists():
+        await message.err("`Seems that an invalid file path provided?`")
+        return
+    return file_path, file_name
+
+
+@userge.on_cmd("x256", about={
+    'header': "Encode a file using x256",
+    'options': {'-b': 'Custom bitrate'},
+    'usage': "{tr}x256 [file / folder path | direct link | reply to telegram file]",
+    'examples': ['{tr}x256 link', '{tr}x256 path']}, check_downpath=True)
+async def encode_x256(message: Message):
+    replied = message.reply_to_message
+    if not message.filtered_input_str and not hasattr(replied, 'media'):
+        await message.edit("`Please read .help x256`")
+        return
+    custom_bitrate = 28
+    match = re.search(r'(\d+)? ?([\s\S]*)?', message.filtered_input_str)
+    if 'b' in message.flags and match:
+        custom_bitrate = match.group(1) if match else "28"
+    dl_loc, file_name = await get_media_path_and_name(message, input_str=match.group(2))
+    if not dl_loc or not file_name:
+        return
+    FF_MPEG_DOWN_LOAD_MEDIA_PATH.mkdir(parents=True, exist_ok=True)
+    video_file = f"{FF_MPEG_DOWN_LOAD_MEDIA_PATH}/x256_{file_name}"
+    await message.edit("`Encoding to x256...`")
+    start = datetime.now()
+    try:
+        ffmpeg.input(dl_loc).output(
+            video_file, vcodec="libx265", crf=custom_bitrate, preset="ultrafast"
+        ).overwrite_output().run(capture_stdout=True, capture_stderr=True)
```

Comment:
you should use run_in_thread, I think 🤔

Reply:
Run function is from python ffmpeg package, not python stdlib.

Reply:
Alright, there are run_async in ffmpeg package, I will check it.

Reply:
both run and run_async are using subprocess: https://github.com/kkroening/ffmpeg-python/blob/master/ffmpeg/_run.py#L288

Reply:
I meant https://github.com/UsergeTeam/Userge/blob/alpha/userge/core/ext/pool.py#L30

Reply:
Handled it in different way, check the last commit.